### ISSUE TITLE
Add "Remove from Scratchpad" to conditions list.

### ISSUE THREAD

**Issue by kaaniboy** [HII-C/patient_viewer#44]

**Comment by kaaniboy** [comment]
You can add the `columnState` of the Conditions list to the conditions service. Then, in the `ConditionsDisplay` component, you can conditionally add either the "Add to ScratchPad" or "Remove from ScratchPad" item to the menu (based on the `columnState` in the service).

**Comment by mattbudiman** [comment]
I think we were, for the most part, able to successfully implement this functionality; however, I'm not entirely sure if our solution is 100% valid. The major bug we faced was that the context menu would display the correct options and perform the correct behavior only after toggling the column state (e.g. switching from default view to scratch pad view) at least once. I think this is because when the app is inititally loaded, the function call `this.conditionService.getColumnState()` returns an undefined value in the following code that we previously wrote in the conditions display component:
```TypeScript
if (this.conditionService.getColumnState() === "default") {
    // Add "Add to Scratch Pad" option to context menu.
} else {
    // Add "Remove from Scratch Pad" option to context menu.
}
```
To fix this error, I simply changed the `if` statement to check for the scratch pad column state instead:
```TypeScript
if (this.conditionService.getColumnState() === "scratchpad") {
    // Add "Remove from Scratch Pad" option to context menu.
} else {
    // Add "Add to Scratch Pad" option to context menu.
}
```
Basically, if `this.conditionService.getColumnState()` returns a value other than `"scratchpad"`, the context menu defaults to the "Add to Scratch Pad" option. This solution works if we assume that the user always starts the application with the default column state. However, what if the user wants to start the application from the scratch pad column state? Then this solution would fail for the same reason as stated above. Is this something we need to account for in our fix? If so, I believe we would need to find a way to initialize the column state in the condition service to the correct state right when the user first opens the application.

**Comment by kaaniboy** [comment]
That's a smart fix to the issue! I wouldn't worry too much about the application starting from the scratch pad state by default. But yes, your idea on how to solve it seems reasonable to me. Good work guys!
